cmd/lvh/runner/runner.go

```diff
@@ -0,0 +1,305 @@
+				"mount flag '%s' doesn't contain a id, must be <id>:<hostpath> or <id>:<hostpath>:<vmpath>",
+				flag,
+			)
+		}
+
+		hostPath, vmPath, found := strings.Cut(paths, ":")
+		if !found {
+			hostPath = paths
+			vmPath = paths
+		}
+
+		if strings.HasPrefix(hostPath, "~") || strings.HasPrefix(vmPath, "~") {
+			homedir, err := os.UserHomeDir()
+			if err != nil {
+				return nil, err
+			}
+
+			hostPath = strings.Replace(hostPath, "~", homedir, 1)
+			vmPath = strings.Replace(vmPath, "~", homedir, 1)
+		}
+
+		var err error
+		hostPath, err = filepath.Abs(hostPath)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"mount flag '%s': %w",
+				flag,
+				err,
+			)
+		}
+
+		vmPath, err = filepath.Abs(vmPath)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"mount flag '%s': %w",
+				flag,
+				err,
+			)
+		}
+
+		qfs = append(qfs, &VirtIOFilesystem{
+			ID:      id,
+			Hostdir: hostPath,
+			VMdir:   vmPath,
+		})
+	}
+
+	return qfs, nil
+}
+
+func parsePorts(flags []string) ([]PortForward, error) {
+	var forwards []PortForward
+	for _, flag := range flags {
+		hostPortStr, vmPortAndProto, found := strings.Cut(flag, ":")
+		if !found {
+			hostPort, err := strconv.Atoi(flag)
+			if err != nil {
+				return nil, fmt.Errorf("'%s' is not a valid port number", flag)
+			}
+			forwards = append(forwards, PortForward{
+				HostPort: hostPort,
+				VMPort:   hostPort,
+				Protocol: "tcp",
+			})
+			continue
+		}
+
+		hostPort, err := strconv.Atoi(hostPortStr)
+		if err != nil {
+			return nil, fmt.Errorf("'%s' is not a valid port number", hostPortStr)
+		}
+
+		vmPortStr, proto, found := strings.Cut(vmPortAndProto, ":")
+		if !found {
+			vmPort, err := strconv.Atoi(vmPortAndProto)
+			if err != nil {
+				return nil, fmt.Errorf("'%s' is not a valid port number", vmPortAndProto)
+			}
+			forwards = append(forwards, PortForward{
+				HostPort: hostPort,
+				VMPort:   vmPort,
+				Protocol: "tcp",
+			})
+			continue
+		}
+
+		vmPort, err := strconv.Atoi(vmPortStr)
+		if err != nil {
+			return nil, fmt.Errorf("'%s' is not a valid port number", vmPortStr)
+		}
+
+		proto = strings.ToLower(proto)
+		if proto != "tcp" && proto != "udp" {
+			return nil, fmt.Errorf("port forward protocol must be tcp or udp")
+		}
+
+		forwards = append(forwards, PortForward{
+			HostPort: hostPort,
+			VMPort:   vmPort,
+			Protocol: proto,
+		})
+	}
+
+	return forwards, nil
+}
+
+var ciliumTesterService = `
+[Unit]
+Description=Cilium tester
+After=network.target docker.service
+
+[Service]
+ExecStart=%s
+Type=oneshot
+# https://www.freedesktop.org/software/systemd/man/systemd.exec.html
+StandardOutput=file:%s
+# StandardOutput=tty
+# StandardOutput=journal+console
+
+[Install]
+WantedBy=multi-user.target
+`
+
+const CiliumTestExecPath = "/usr/bin/cilium-test"
+
+func BuildTesterService(rcnf *RunConf, tmpDir, scriptPath, scriptOutput string) ([]images.Action, error) {
+	service := fmt.Sprintf(ciliumTesterService, CiliumTestExecPath, scriptOutput)
```

Comment:
Can we have the service file as an argument to this global function? we can use the ciliumTestService (maybe rename it?) as the default of no service file is provided by the caller.

Reply:
I got rid of the service completely, as for now we are going to run tests via SSH.